packages/worker/src/api/routes/global/tests/scim/groups.spec.ts

```diff
@@ -0,0 +1,183 @@
+import tk from "timekeeper"
+import _ from "lodash"
+import { mocks, structures } from "@budibase/backend-core/tests"
+import { ScimCreateGroupRequest, ScimGroupResponse } from "@budibase/types"
+import { TestConfiguration } from "../../../../../tests"
+
+mocks.licenses.useScimIntegration()
+
+function createScimCreateGroupRequest(groupData?: {
+  externalId?: string
+  displayName?: string
+}) {
+  const {
+    externalId = structures.uuid(),
+    displayName = structures.generator.word(),
+  } = groupData || {}
+
+  const group: ScimCreateGroupRequest = {
+    schemas: [
+      "urn:ietf:params:scim:schemas:core:2.0:Group",
+      "http://schemas.microsoft.com/2006/11/ResourceManagement/ADSCIM/2.0/Group",
+    ],
+    externalId: externalId,
+    displayName: displayName,
+    meta: {
+      resourceType: "Group",
+    },
+  }
+  return group
+}
+
+describe("/api/global/scim/v2/groups", () => {
+  let mockedTime = new Date(structures.generator.timestamp())
+
+  beforeEach(() => {
+    tk.reset()
+    mockedTime = new Date(structures.generator.timestamp())
+    tk.freeze(mockedTime)
```

Comment:
we have a constant mocked timestamp already that could be used instead? it's applied by default for all tests so could remove this new freezing and import the constant from structures instead. or is there another reason for using a new timestamp for each test?

I'd like to keep data handling as as noninvasive as possible which is why I always question date manipulation in individual tests

Reply:
Update it to use the generic mocked time and overriding it when actually needed